server/src/main/java/org/elasticsearch/index/mapper/BooleanScriptFieldType.java

```diff
@@ -67,14 +67,21 @@ public static RuntimeField sourceOnly(String name) {
         return new Builder(name).createRuntimeField(BooleanFieldScript.PARSE_FROM_SOURCE);
     }
 
-    BooleanScriptFieldType(String name, BooleanFieldScript.Factory scriptFactory, Script script, Map<String, String> meta) {
-        super(
-            name,
-            searchLookup -> scriptFactory.newFactory(name, script.getParams(), searchLookup),
-            script,
-            scriptFactory.isResultDeterministic(),
-            meta
-        );
+    BooleanScriptFieldType(
+        String name,
+        BooleanFieldScript.Factory scriptFactory,
+        Script script,
+        Map<String, String> meta,
+        boolean onErrorContinue
+    ) {
+        super(name, searchLookup -> {
+            BooleanFieldScript.LeafFactory leafFactory = scriptFactory.newFactory(name, script.getParams(), searchLookup);
```

Comment:
I think I would have taken a slightly different approach: add an argument to newFactory after searchLookup. as well as an argument to all script classes constructor. That would allow you to make the option final in the script and there is no need for a setter method. The function that creates the script does not need to be customized also.